### PR TITLE
improvement(k8s-eks): cleanup AWS launch templates created by EKS

### PR DESCRIPTION
**Pull request by vponomaryov**
Recently we started using `launch template` resources in EKS deployments.
And we don't delete it after test runs.

So, add cleanup code for that kind of resources.

### Testing

- [ ]

### PR pre-checks (self review)

- [ ] I added the relevant `backport` labels
- [ ] I didn't leave commented-out/debugging code

### Reminders

- Add New configuration option and document them (in `sdcm/sct_config.py`)
- Add unit tests to cover my changes (under `unit-test/` folder)
- Update the Readme/doc folder relevent to this change (if needed)

### CODE REVIEW

**Comment by vponomaryov** [comment]
Tested here: https://jenkins.scylladb.com/job/scylla-staging/job/valerii/job/vp-functional-eks/67/consoleFull

**Comment by vponomaryov** [comment]
The PR's CI EKS job failed to provision resources:
```
19:55:00  < t:2024-01-31 17:54:55,269 f:file_logger.py  l:101  c:sdcm.sct_events.file_logger p:ERROR > 2024-01-31 17:54:55 [✖]  failed to create service account kube-system/functional-PR-7174-832d6dde: ServiceAccount "functional-PR-7174-832d6dde" is invalid: metadata.name: Invalid value: "functional-PR-7174-832d6dde": a lowercase RFC 1123 subdomain must consist of lower case alphanumeric characters, '-' or '.', and must start and end with an alphanumeric character (e.g. 'example.com', regex used for validation is '[a-z0-9]([-a-z0-9]*[a-z0-9])?(\.[a-z0-9]([-a-z0-9]*[a-z0-9])?)*')
```

Failure is not related to this PR.

File: sdcm/utils/common.py [hunk 1]
```diff
@@ -607,6 +607,7 @@ def clean_cloud_resources(tags_dict, config=None, dry_run=False):  # pylint: dis
         return
     if cluster_backend in ('k8s-eks', ''):
         clean_clusters_eks(tags_dict, regions=aws_regions, dry_run=dry_run)
+        clean_launch_templates_aws(tags_dict, regions=aws_regions, dry_run=dry_run)
```
Comment (fruch) [inline sdcm/utils/common.py:610]: Nitpick: it would nice if list resources could show it

**Review by fruch — APPROVED** [review]
LGTM

**Pull request merged**